### PR TITLE
enable operator

### PR DESCRIPTION
**Pull request by sanchezl**

### CODE REVIEW

File: pkg/operator/targetcontroller/controller.go [hunk 1]
```diff
@@ -58,16 +63,26 @@ func NewTargetController(kubeClient kubernetes.Interface,
 	}
 
 	genericOperatorConfigClient.Informer().AddEventHandler(controller.eventHandler())
+	secretInformer.Informer().AddEventHandler(controller.eventHandler())
+	deploymentInformer.Informer().AddEventHandler(controller.eventHandler())
+
+	controller.cachesToSync = append(controller.cachesToSync,
+		genericOperatorConfigClient.Informer().HasSynced,
+		deploymentInformer.Informer().HasSynced,
```
Comment (sttts) [inline pkg/operator/targetcontroller/controller.go]: why not secretInformer?

Comment (sanchezl) [reply pkg/operator/targetcontroller/controller.go]: Fixed

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -61,10 +62,15 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 	}
 	versionRecorder.SetVersion("operator", os.Getenv("OPERATOR_IMAGE_VERSION"))
 
+	kubeInformersForTargetNamespace := informers.NewSharedInformerFactoryWithOptions(kubeClient, 10*time.Minute,
```
Comment (sttts) [inline pkg/operator/starter.go]: we have `v1helpers.KubeInformersForNamespaces`. Will simplify this.

File: pkg/operator/targetcontroller/sync.go [hunk 3]
```diff
@@ -57,14 +57,20 @@ func (c *TargetController) syncKubeStorageVersionManager(spec *operatorv1.KubeSt
 		c.versionRecorder.SetVersion("kube-storage-version-migrator", operandVersion)
 	}
 
-	_, _, err = v1helpers.UpdateStatus(c.genericOperatorConfigClient, func(oldStatus *operatorv1.OperatorStatus) error {
-		for _, condition := range operatorStatus.Conditions {
-			if err := v1helpers.UpdateConditionFn(condition)(oldStatus); err != nil {
-				return err
-			}
-		}
+	var updateStatusFuncs []v1helpers.UpdateStatusFunc
```
Comment (sttts) [inline pkg/operator/targetcontroller/sync.go]: // patch conditions, observedGenerations and Generation in the existing operation status

**Comment by sttts** [comment]
/hold

waiting for the image https://jira.coreos.com/browse/ART-1227.

/lgtm
/approve

File: bindata/kube-storage-version-migrator/deployment.yaml [hunk 4]
```diff
@@ -20,7 +20,10 @@ spec:
...
         command:
-          - /migrator
+          - migrator
           - '--alsologtostderr'
         terminationMessagePolicy: FallbackToLogsOnError
-        imagePullPolicy: Always
+        resources:
+            requests:
+              cpu: 10m
+              memory: 50Mi
```
Comment (sttts) [inline bindata/kube-storage-version-migrator/deployment.yaml]: is this really enough? Doesn't sound much.

Comment (sttts) [reply bindata/kube-storage-version-migrator/deployment.yaml]: In theory one object could be 2 MB in size. 500 are listed per request => 1 GB of RAM. On the other hand, I am not sure we want to guarantee 1 GB for a process that is idle most of the time.

**Pull request merged**